### PR TITLE
Improve cat Command

### PR DESCRIPTION
**Pull request by mmstick**
- Implemented all flag arguments except -v, as I literally have no idea what -v does.

### CODE REVIEW

File: src/bin/cat.rs [hunk 1]
```diff
@@ -3,30 +3,207 @@
...
+                        if byte == b'\n' {
+                            if current_line.is_empty() {
+                                if last_line_was_blank && self.squeeze_blank {
+                                    continue
+                                } else if !last_line_was_blank {
+                                    last_line_was_blank = true;
+                                }
+                            }
+                            if self.number || (self.number_nonblank && !current_line.is_empty()) {
+                                stdout.write(format!("     {}  ", line_count).as_bytes()).try(stderr);
```
Comment (Ticki) [inline src/bin/cat.rs]: No need for formatters.

File: src/bin/cat.rs [hunk 2]
```diff
@@ -3,30 +3,207 @@
...
+        for arg in env::args().skip(1) {
+            if arg.starts_with('-') {
+                match arg.as_str() {
+                    "-h" | "--help" => {
+                        stdout.write(MAN_PAGE.as_bytes()).try(stderr);
+                        stdout.flush().try(stderr);
+                        std::process::exit(0);
+                    }
+                    "-A" | "--show-all" => {
+                        // TODO: implement -v
```
Comment (Ticki) [inline src/bin/cat.rs]: `-v` simply notates non-printable characters too, e.g.`^I`.

File: src/bin/cat.rs (hunk 1 above)
Comment (mmstick) [inline src/bin/cat.rs]: I'm unaware of any other method of converting a number into a string.

Comment (Ticki) [inline src/bin/cat.rs]: `a.to_string()`.

File: src/bin/cat.rs (hunk 2 above)
Comment (mmstick) [inline src/bin/cat.rs]: Doesn't seem to print `^I` which is a tab. I see no difference in the output of cat -v versus cat.

Comment (Ticki) [inline src/bin/cat.rs]: Try to run `cat -v` on a binary file, like an executable or an image.

**Pull request merged**